### PR TITLE
Rename TypeHelper to ITypeSymbolExtensions

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
There are some file-scoped namespaces and file renames.  Review per commit should be easier

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.Common/Extensions/ITypeSymbolExtensions.cs [hunk 1]
```diff
@@ -112,20 +106,6 @@ public static bool IsAny(this ITypeSymbol typeSymbol, ImmutableArray<KnownType>
...
-        parameter != null && parameter.Type.Is(type);
-
-    public static bool IsInType(this ISymbol symbol, KnownType type) =>
-        symbol != null && symbol.ContainingType.Is(type);
-
-    public static bool IsInType(this ISymbol symbol, ITypeSymbol type) =>
-        symbol?.ContainingType != null && symbol.ContainingType.Equals(type);
-
-    public static bool IsInType(this ISymbol symbol, ImmutableArray<KnownType> types) =>
-        symbol != null && symbol.ContainingType.IsAny(types);
```
Comment (sebastien-marichal) [inline analyzers/src/SonarAnalyzer.Common/Extensions/ITypeSymbolExtensions.cs:125]: New styling rule idea: Methods in an extension class should extend the same type.

File: analyzers/src/SonarAnalyzer.Common/Extensions/IParameterSymbolExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Extensions;
+
+internal static class IParameterSymbolExtensions
+{
+    public static bool IsType(this IParameterSymbol parameter, KnownType type) =>
+        parameter is not null && parameter.Type.Is(type);
```
Comment (sebastien-marichal) [inline analyzers/src/SonarAnalyzer.Common/Extensions/IParameterSymbolExtensions.cs:26]: We could add unit test for this extensions class.

File: analyzers/src/SonarAnalyzer.Common/Extensions/ISymbolExtensions.cs [hunk 3]
```diff
@@ -20,58 +20,66 @@
...
+            : method.ContainingType.Is(containingType);
+
+    public static bool IsInType(this ISymbol symbol, KnownType type) =>
+        symbol is not null && symbol.ContainingType.Is(type);
+
+    public static bool IsInType(this ISymbol symbol, ITypeSymbol type) =>
+        symbol?.ContainingType is not null && symbol.ContainingType.Equals(type);
+
+    public static bool IsInType(this ISymbol symbol, ImmutableArray<KnownType> types) =>
+        symbol is not null && symbol.ContainingType.IsAny(types);
```
Comment (sebastien-marichal) [inline analyzers/src/SonarAnalyzer.Common/Extensions/ISymbolExtensions.cs:84]: Coverage could be easily improved by adding dedicated tests in the UTs.

**Review by sebastien-marichal — CHANGES_REQUESTED** [review]
100% coverage can be easily reached.
All missing coverages are about `null` value.

File: analyzers/src/SonarAnalyzer.Common/Extensions/ITypeSymbolExtensions.cs (hunk 1 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/src/SonarAnalyzer.Common/Extensions/ITypeSymbolExtensions.cs:125]: #9227

**Review by sebastien-marichal — APPROVED** [review]
LGTM!

**Pull request merged**